src/query/sharing/src/share_endpoint.rs

```diff
@@ -13,50 +13,170 @@
 // limitations under the License.
 
 use std::collections::BTreeMap;
+use std::sync::Arc;
+use std::sync::RwLock;
```

Comment:
Use `use parking_lot::RwLock` instead? This we can not define a new error code:`RwLockError(1112),`